bdb/cursor.c

```diff
@@ -7956,6 +7956,15 @@ static int update_pglogs_from_global_queues_int(
 
         if (!found_greater)
             current = NULL;
+
+        while (current && current != last) {
+            current = current->lnk.next;
+            update_pglogs_from_queue(cur->shadow_tran, qcur->fileid, current);
```

Comment:
Is there a chance current can be NULL here, should we check?

Reply:
It shouldn't happen- it would have to go past "last" to be NULL, which was set under the same readlock that we're currently holding.  The thread which cleans these up holds the write-lock.

Reply:
If it's NULL I want to see the core that this will produce

Reply:
Ok, we first go to prev elements, then we walk next and we stop when we hit current == last. 
This is fine.